frontend/src/components/InputFormMeetings.tsx

```diff
@@ -0,0 +1,140 @@
+import React, {useState} from "react";
+import {Button, Dialog, DialogActions, DialogContent, DialogContentText, TextField} from "@mui/material";
+import {useNavigate} from "react-router-dom";
+import {styled} from "styled-components";
+import CheckIcon from "@mui/icons-material/Check";
+import CloseIcon from "@mui/icons-material/Close";
+
+type Props = {
+    title: string;
+    initialFormData: MeetingWithoutId;
+    onCancel: () => void;
+    onSubmit: (formData: MeetingWithoutId) => void;
+};
+
+export default function InputFormMeetings({ title, initialFormData, onCancel, onSubmit }: Props) {
+    const [formData, setFormData] = useState<MeetingWithoutId>(initialFormData);
+    const [open, setOpen] = useState(false);
+    const navigate = useNavigate();
+
+    function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
+        const name = event.target.name;
+        const value = event.target.value;
+        setFormData((prevData) => ({ ...prevData, [name]: value }));
+    }
+
+    function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+        event.preventDefault();
+        onSubmit(formData);
+        navigate("/meetinglist");
+    }
+
+    const handleCancel = () => {
+        setOpen(true);
+    };
+
+    const handleClose = () => {
+        setOpen(false);
+    };
+
+    const handleConfirmCancel = () => {
+        setOpen(false);
+        onCancel();
+    }
+
+    return (
+        <form onSubmit={handleSubmit} style={{ display: "flex", flexDirection: "column", alignItems: "center" }}>
+            <h1 style={{ margin: "20px 0" }}>{title}</h1>
+            <TextField
```

Comment:
Here I would also think about a "combined" styling component for the Textfields & Buttons on the end of the file instead of in the component itself

Reply:
thank you, I'll try that for sure!